app/services/github_client/base.rb

```diff
@@ -1,11 +1,29 @@
 module GithubClient
   class Base
+    private
+
     def connection
       Faraday.new('https://api.github.com') do |conn|
         conn.basic_auth(ENV['GITHUB_ADMIN_USER'], ENV['GITHUB_ADMIN_TOKEN'])
         conn.headers['Accept'] = 'application/vnd.github.v3+json'
         conn.response :raise_error
       end
     end
+
+    def get_all_paginated_items(url, max_per_page, accumulated_items = [], page_number = 1)
```

Comment:
minor comment but maybe `max_per_page, accumulated_items = [], page_number = 1` should be keyword arguments because they look more like options than to ordinary arguments to me.. thoughts? also max_per_page could have a default value too, right?

Reply:
I think you may be right about the keyword arguments.
As for the default value, it's not that straightforward. Each API could have it's own max value, so placing a default value would assume they are all the same ones, which [Github specifically says it may not be the case](https://docs.github.com/en/rest/guides/traversing-with-pagination). So I'd rather leave that one as it is.